lib/sparql.ts

```diff
@@ -279,12 +279,14 @@ function translateExtend(op: Algebra.Extend): any
 function translateFrom(op: Algebra.From): GroupPattern
 {
     const result = translateOperation(op.input);
-    // project is nested in group object
-    const obj: SelectQuery = result.patterns[0];
-    obj.from = {
-        default: op.default,
-        named: op.named
-    };
+    if (result.patterns[0] !== undefined) {
```

Comment:
Does this issue occur when the first entry of the `patterns` array is `undefined`, or if the `patterns` array is empty, which causes this to result to be `undefined`? If it is the latter, this check should be `result.patterns.length > 0`.

Reply:
Related to this, might make sense to add a unit test for it (or a query in which the problem used to occur).

Reply:
> Does this issue occur when the first entry of the `patterns` array is `undefined`, or if the `patterns` array is empty.
I mistyped my check. The issue occurs when the patterns array is undefined I've updated the if condition.

Reply:
> might make sense to add a unit test for it (or a query in which the problem used to occur).
To set initialbindings for the test, should I add another function in sparql-test.js?

For completeness, the case which failed was:
```
query = CONSTRUCT FROM <http://www.ontotext.com/retain-bind-position> WHERE { ?s ?p ?o }  LIMIT 100
const quadStream = await this.engine.queryQuads(query, {
  sources: [
        {
          type: 'sparql',
          value: 'https://www.goudatijdmachine.nl/sparql/repositories/gtm',
        },
      ],
  initialBindings: bindingsFactory.fromRecord({}) as unknown as Bindings,
});
```

Reply:
You need to add the SPARQL query somewhere in the `test/sparql` folder and the expected algebra result in the same location in the other 2 test folders. During unit tests it is checked if the query correctly converts to the algebra and if the algebra correctly converts to the query.